### PR TITLE
replacement and deprecation of margins()

### PR DESCRIPTION
**Pull request by gabrielodom**
Work with @gralgomez on Issue #85; added `test_deprecated.R` for future deprecation warning tests

### CODE REVIEW

File: tests/testthat/test_parameters.R [hunk 1]
```diff
@@ -19,7 +19,10 @@ test_that('param ranges', {
   expect_equal(num_comp(1:2)$range, list(lower = 1L, upper = 2L))
   expect_equal(cost(-2:-1)$range, list(lower = -2, upper = -1))
   expect_equal(scale_factor(-2:-1)$range, list(lower = -2, upper = -1))
-  expect_equal(dials::margin(c(.1, .15))$range, list(lower = .1, upper = .15))
+  expect_warning(
```
Comment (DavisVaughan) [inline tests/testthat/test_parameters.R]: Can you please pull out this test and move it to `test_deprecated.R` ?

File: R/param_svm.R [hunk 2]
```diff
@@ -23,6 +23,25 @@ cost <- function(range = c(-10, -1), trans = log2_trans()) {
 #' @rdname cost
 #' @export
 margin <- function(range = c(0, .2), trans = NULL) {
+  warning(
+    "The margin() function will be deprecated in versions >= 0.0.6.9000. Please
+    use svm_margin() instead. See https://github.com/tidymodels/dials/issues/85
+    for more details."
+  )
+
+  new_quant_param(
```
Comment (DavisVaughan) [inline R/param_svm.R]: To be consistent, you can just call `svm_margin(range = range, trans = trans)`

File: tests/testthat/test_deprecated.R [hunk 3]
```diff
@@ -0,0 +1,12 @@
+library(testthat)
+library(dials)
+
+context("test for deprecation warnings")
+
+test_that("deprecation warnings appear", {
```
Comment (DavisVaughan) [inline tests/testthat/test_deprecated.R]: I think you can remove this test in favor of the one that would come from `test_parameters.R`

File: tests/testthat/test_deprecated.R [hunk 4]
```diff
@@ -0,0 +1,12 @@
+library(testthat)
+library(dials)
```
Comment (DavisVaughan) [inline tests/testthat/test_deprecated.R]: no need to library these

**Comment by DavisVaughan** [comment]
Once you finish, I'll also change your `warning()` call to use `lifecycle::deprecate_soft()` instead. Feel free to read the documentation and go ahead and make that change. It would also require importing lifecycle with `usethis::use_package("lifecycle")`

File: tests/testthat/test_deprecated.R (hunk 3 above)
Comment (gabrielodom) [reply tests/testthat/test_deprecated.R]: Spoke with @DavisVaughan ; will leave.

File: NEWS.md [hunk 5]
```diff
@@ -1,5 +1,7 @@
 # dials (development version)
 
+* Renamed `margin()` to `svm_margin()` (#85). We've updated the tests for `margin()` and `svm_margin()` and added the `test_deprecated.R` test script. (@gabrielodom and @gralgomez, #85)
```
[the remark this answers is not in the document]
Comment (gralgomez) [reply NEWS.md]: Gotcha! Complete

File: tests/testthat/test_parameters.R (hunk 1 above)
Comment (gralgomez) [reply tests/testthat/test_parameters.R]: Done. Also `dials::` from previous version was removed

File: R/param_svm.R (hunk 2 above)
Comment (gralgomez) [reply R/param_svm.R]: @DavisVaughan do you mean only calling `svm_margins` instead of having both functions (margins / svm_margins) within the file?

Comment (gabrielodom) [reply R/param_svm.R]: Our update:
```
margin <- function(range = c(0, .2), trans = NULL) {
  warning(
    "The margin() function will be deprecated in versions >= 0.0.6.9000. Please
    use svm_margin() instead. See https://github.com/tidymodels/dials/issues/85
    for more details."
  )
  
  svm_margin(range = range, trans = trans)
}
```

File: tests/testthat/test_deprecated.R (hunk 4 above)
Comment (gralgomez) [reply tests/testthat/test_deprecated.R]: `testhat`/`dials` library calls removed from `test_deprecated.R` and all other test files.

File: R/param_svm.R (hunk 2 above)
Comment (DavisVaughan) [reply R/param_svm.R]: ^ this looks great

**Comment by DavisVaughan** [comment]
Thanks for the hard work!

**Pull request merged**